api/v1beta1/ipaddresspool_types.go

```diff
@@ -49,6 +49,14 @@ type IPAddressPoolSpec struct {
 	// multiple IPAddressPools have the same priority, choice will be random.
 	// +optional
 	AllocateTo *ServiceAllocation `json:"serviceAllocation,omitempty"`
+
+	// DisablePoolSelection disables ip pool selection when its ip addresses
+	// are requested via service's metallb.universe.tf/loadBalancerIPs or
+	// metallb.universe.tf/address-pool annotations or Spec.LoadBalancerIP
+	// field.
+	// +optional
+	// +kubebuilder:default:=false
+	DisablePoolSelection bool `json:"disablePoolSelection,omitempty"`
```

Comment:
I start to think we don't need this anymore (see comment below) because we check if a service is asking for an ip (or a pool) it's allowed to use